src/main/php/org/bovigo/vfs/visitor/vfsStreamPrintVisitor.php

```diff
@@ -40,12 +39,14 @@ class vfsStreamPrintVisitor extends vfsStreamAbstractVisitor
      *
      * If no file pointer given it will fall back to STDOUT.
      *
-     * @param   resource  $out  optional
+     * @param   resource|null  $out  optional
      * @throws  \InvalidArgumentException
      * @api
      */
-    public function __construct($out = STDOUT)
+    public function __construct($out = null)
     {
+        $out = $out ?? STDOUT;
+
```

Comment:
Why is it better to check and assign STDOUT than simply let the default value be STDOUT?

Reply:
It's not necessarily better, it's only to fix what phpstan sees as an error. For whatever reason, phpstan sees `STDOUT` as having the type `mixed`, even though it should only ever be a `resource` or `false` (highly unlikely). Because of this, it throws an error saying `Default value of the parameter #1 $out (mixed) of method org\bovigo\vfs\visitor\vfsStreamPrintVisitor::__construct() is incompatible with type resource.`

An alternate fix would be to change the doc block to say it accepts `mixed`, but that seemed more wrong since it's intended to only accept a resource.

Reply:
If it doesn't improve the code I think we shouldn't make it slightly worse just to appease a tool. I'd rather add this to an `ignoreErrors` section in the phpstan config file.

Reply:
Changed it back and added to the `ignoreErrors` section